tests/test_cms_menus.py

```diff
@@ -82,33 +82,73 @@ def test_get_nodes(self):
+            nodes[0],
+            title="",
+            url="",
+            id=menu_versions[0].content.menu.root_id,
+            parent_id=None,
+            attr={},
+        )
+        self.assertNavigationNodeEqual(
+            nodes[1],
+            title="",
+            url="",
+            id=menu_versions[1].content.menu.root_id,
+            parent_id=None,
+            attr={},
+        )
+        self.assertNavigationNodeEqual(
+            nodes[2],
+            title=child1.title,
+            url=child1.content.get_absolute_url(),
+            id=child1.id,
+            parent_id=menu_versions[0].content.menu.root_id,
+            attr={"link_target": child1.link_target},
+        )
+        self.assertNavigationNodeEqual(
+            nodes[3],
+            title=grandchild.title,
+            url=grandchild.content.get_absolute_url(),
+            id=grandchild.id,
+            parent_id=child1.id,
+            attr={"link_target": grandchild.link_target},
+        )
+        self.assertNavigationNodeEqual(
+            nodes[4],
+            title=child2.title,
+            url=child2.content.get_absolute_url(),
+            id=child2.id,
+            parent_id=menu_versions[1].content.menu.root_id,
+            attr={"link_target": child2.link_target},
+        )
+
+    def test_get_roots_with_draft_mode(self):
         """This test to check versioning would group all the versions
         of menu content and return latest of all distinct menu content
         """
         menucontent_1_v1 = factories.MenuVersionFactory(state=ARCHIVED)
-        # draft version
         menucontent_1_v2 = factories.MenuVersionFactory(
-            content__menu=menucontent_1_v1.content.menu
+            content__menu=menucontent_1_v1.content.menu, state=DRAFT
         )
-        # draft version
-        menucontent_2_v1 = factories.MenuVersionFactory()
-        # assert to check draft_mode_active is false
+        menucontent_2_v1 = factories.MenuVersionFactory(state=PUBLISHED)
+        menucontent_3_v1 = factories.MenuVersionFactory(state=UNPUBLISHED)
+        # Assert to check draft_mode_active is false
         self.assertFalse(self.menu.renderer.draft_mode_active)
         roots = self.menu.get_roots(self.request)
 
-        # renderer should only render published menucontent
-        # In this testcase there isnt any published version
-        self.assertEqual(roots.count(), 0)
-
-    def test_get_roots_when_draft_mode_active(self):
-        menucontent_1_v1 = factories.MenuVersionFactory(state=ARCHIVED)
-        # draft version
-        menucontent_1_v2 = factories.MenuVersionFactory(
-            content__menu=menucontent_1_v1.content.menu
-        )
-        # draft version
-        menucontent_2_v1 = factories.MenuVersionFactory()
+        # Renderer should only render published menucontent
+        self.assertEqual(roots.count(), 1)
+        self.assertListEqual(list(roots), [menucontent_2_v1.content.root])
```

Comment:
I think draft mode active is True is one test and draft mode active is False is another test (as it was before). The test I think you can remove now that you've added unpublished versions is `test_get_roots_for_archived_and_unpublished_versions`